Library/Homebrew/bundle/services.rb

```diff
@@ -0,0 +1,92 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "formula"
+require "services/system"
+
+module Homebrew
+  module Bundle
+    module Services
+      sig {
+        params(
+          entries: T::Array[Homebrew::Bundle::Dsl::Entry],
+          _block:  T.proc.params(info: T::Hash[String, T.anything], service_file: Pathname).void,
+        ).void
+      }
+      private_class_method def self.map_entries(entries, &_block)
+        formula_versions = {}
+        ENV.each do |key, value|
+          match = key.match(/^HOMEBREW_BUNDLE_EXEC_FORMULA_VERSION_(.+)$/)
+          next if match.blank?
+
+          formula_name = match[1]
+          next if formula_name.blank?
+
+          ENV.delete(key)
+          formula_versions[formula_name.downcase] = value
+        end
+
+        entries.filter_map do |entry|
+          next if entry.type != :brew
+
+          formula = Formula[entry.name]
+          next unless formula.any_version_installed?
+
+          version = formula_versions[entry.name.downcase]
+          prefix = formula.rack/version if version
+
+          service_file = if prefix&.directory?
+            if Homebrew::Services::System.launchctl?
+              prefix/"#{formula.plist_name}.plist"
+            else
+              prefix/"#{formula.service_name}.service"
+            end
+          end
+
+          unless service_file&.file?
+            prefix = formula.any_installed_prefix
+            next if prefix.nil?
+
+            service_file = if Homebrew::Services::System.launchctl?
+              prefix/"#{formula.plist_name}.plist"
+            else
+              prefix/"#{formula.service_name}.service"
+            end
+          end
+
+          next unless service_file.file?
+
+          # We parse from a command invocation so that brew wrappers can invoke special actions
+          # for the elevated nature of `brew services`
+          output = Utils.safe_popen_read(HOMEBREW_BREW_FILE, "services", "info", "--json", formula.full_name)
```

Comment:
Does `brew services list --json` contain the information we need? If so, would be nice to use that instead. If not, maybe it'd be nice to add that information in there.

Reply:
`list` is a small subset of `info`. I use 4 fields from the JSON, 3 of which are not in `list`.

Could maybe add them though we'd be near the point of making it not a subset anymore.

I think maybe a better option is to allow `info` to take in multiple arguments.